subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyConstraintsIntegrationTest.groovy

```diff
@@ -0,0 +1,247 @@
+
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.integtests.resolve
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.ExperimentalFeaturesFixture
+
+/**
+ * This is a variation of {@link OptionalDependenciesIntegrationTest} that tests dependency constraints
+ * declared in the build script (instead of published optional dependencies)
+ */
+class DependencyConstraintsIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
:o: Let's use the new test fixtures here: `AbstractModuleDependencyResolveTest` and `resolve.expectGraph { ... }`

Reply:
This test tests the declaration of constraints in the DSL. Not in published metadata. For the published case, I changed the _OptionalDependenciesIntegrationTest_ to use `AbstractModuleDependencyResolveTest` in #3630:
https://github.com/gradle/gradle/blob/37a55aeff016b20106c87633854b252e09f13b24/subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/PublishedDependencyConstraintsIntegrationTest.groovy

For the test here:
I did some cleanup and used the  `resolve.expectGraph { ... }` fixture. By that, I actually discovered another issue: https://github.com/gradle/gradle/pull/3588/commits/570d793c579bc759c4f9a4f763c073973b610b2d